packages/@divvi/mobile/src/earn/EarnWithdrawConfirmationScreen.tsx

```diff
@@ -0,0 +1,171 @@
+    return params.mode === 'withdraw'
+      ? new BigNumber(params.inputTokenAmount)
+      : getEarnPositionBalanceValues({ pool: params.pool }).poolBalanceInDepositToken
+  }, [params])
+
+  const localAmount = useTokenToLocalAmount(tokenAmount, depositToken?.tokenId)
+  return { tokenAmount, localAmount, depositToken, withdrawToken }
+}
+
+export default function EarnWithdrawConfirmationScreen({ route: { params } }: Props) {
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+  const providerUrl = params.pool.dataProps.manageUrl ?? params.pool.dataProps.termsUrl
+  const localCurrencySymbol = useSelector(getLocalCurrencySymbol) ?? LocalCurrencySymbol.USD
+  const rewards = useRewards(params)
+  const withdraw = useWithdrawAmountInDepositToken(params)
+
+  if (!withdraw.depositToken || !withdraw.withdrawToken) {
+    // should never happen
+    Logger.error(TAG, 'there is neither deposit nor withdraw token available')
+    return null
+  }
+
+  function onPressProvider() {
+    if (withdraw.withdrawToken) {
+      AppAnalytics.track(EarnEvents.earn_withdraw_provider_info_press, {
+        depositTokenId: params.pool.dataProps.depositTokenId,
+        tokenAmount: withdraw.tokenAmount.toString(),
+        networkId: withdraw.withdrawToken.networkId,
+        providerId: params.pool.appId,
+        poolId: params.pool.positionId,
+        mode: params.mode,
+        rewards: rewards.tokens
+          .filter((token) => !!token.tokenInfo)
+          .map((token) => ({
+            amount: token.balance,
+            tokenId: token.tokenInfo!.tokenId,
+          })),
+      })
+    }
+
+    if (providerUrl) {
+      dispatch(openUrl(providerUrl, true))
+    }
+  }
+
+  return (
+    <ReviewTransaction title={t('earnFlow.withdrawConfirmation.title')}>
+      <ReviewContent>
+        <ReviewSummary>
+          {(params.mode === 'withdraw' || params.mode === 'exit') && (
+            <ReviewSummaryItem
+              testID="EarnWithdrawConfirmation/Withdraw"
+              label={t('earnFlow.withdrawConfirmation.withdrawing')}
+              icon={<TokenIcon token={withdraw.depositToken} />}
+              primaryValue={t('tokenAmount', {
+                tokenAmount: formatValueToDisplay(withdraw.tokenAmount),
+                tokenSymbol: withdraw.depositToken.symbol,
+              })}
+              secondaryValue={t('localAmount', {
+                context: withdraw.localAmount ? undefined : 'noFiatPrice',
+                localAmount: withdraw.localAmount ? formatValueToDisplay(withdraw.localAmount) : '',
+                localCurrencySymbol,
+              })}
+            />
+          )}
+
+          {(params.mode === 'claim-rewards' ||
+            params.mode === 'exit' ||
+            params.pool.dataProps.withdrawalIncludesClaim) &&
+            rewards.tokens.map((rewardToken, idx) => {
+              if (!rewardToken.tokenInfo) return null
+              return (
+                <ReviewSummaryItem
```

Comment:
nit: bit more readable with filter and avoiding the `return`
```suggestion
            rewards.tokens.filter(rewardToken => !!rewardToken.tokenInfo).map((rewardToken, idx) => (
                <ReviewSummaryItem
```